src/demo/demo.js

```diff
@@ -34,8 +34,66 @@ const SplatLink = withRouter(props => {
   }
   return <Link {...newProps}>{props.children}</Link>;
 });
+class EventsListener extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      gotStartLoadingEvent: false,
+      gotComponentReady: false,
+      gotComponentWillUnmount: false
+    };
+  }
+
+  componentWillMount() {
+    if (this.unSubscribeStartLoading) {
+      this.unSubscribeStartLoading();
+    }
+    this.unSubscribeStartLoading = window.ModuleRegistry.addListener('react-module-container.componentStartLoading', () => {
+      this.setState({gotStartLoadingEvent: true});
+    });
+
+    if (this.unSubscribeComponentReady) {
+      this.unSubscribeComponentReady();
+    }
+    this.unSubscribeComponentReady = window.ModuleRegistry.addListener('react-module-container.componentReady', () => {
+      this.setState({gotComponentReady: true});
+    });
+
+    if (this.unSubscribeComponentWillUnmount) {
+      this.unSubscribeComponentWillUnmount();
+    }
+    this.unSubscribeComponentWillUnmount = window.ModuleRegistry.addListener('react-module-container.componentWillUnmount', () => {
+      this.setState({gotComponentWillUnmount: true});
+    });
+  }
+
+  componentWillUnmount() {
+    if (this.unSubscribeStartLoading) {
+      this.unSubscribeStartLoading.remove();
+      this.unSubscribeStartLoading = null;
+    }
+    if (this.unSubscribeComponentReady) {
+      this.unSubscribeComponentReady.remove();
+      this.unSubscribeComponentReady = null;
+    }
+    if (this.unSubscribeComponentWillUnmount) {
+      this.unSubscribeComponentWillUnmount.remove();
+      this.unSubscribeComponentWillUnmount = null;
+    }
+  }
+
+  render() {
+    return (<div>
+      <div>gotStartLoadingEvent: <span id="got-start-loading">{this.state.gotStartLoadingEvent ? 'true' : 'false'}</span></div>
```

Comment:
I've added the label only for the `npm start` so developers will understand what they are seeing. The inner `id` is for the e2e